VMEncryption/main/check_util.py

```diff
@@ -401,17 +401,25 @@ def validate_skr_release_rsa_or_ec_key(self,public_settings,check_release_rsa_or
             return
         manage_id = public_settings.get(CommonVariables.EncryptionManagedIdentity)
         KeyEncryptionKeyUrl=public_settings.get(CommonVariables.KeyEncryptionKeyURLKey)
-        if manage_id:
-            ret = self._update_imds_managed_id_env_variable(manage_id)
-            if not ret:
-                raise Exception(msg)
+        #checking if managed_id and KeyEncryptionKeyUrl is present. mandatory for DDE. 
+        if not manage_id:
+            raise Exception('managed identity is not provided. Managed Identity is a mandatory field.')
+        if not KeyEncryptionKeyUrl:
+            raise Exception('KEK URL is not provided. Key encryption key URL is a mandatory field.')
+        ret = self._update_imds_managed_id_env_variable(manage_id)
+        if not ret:
+            raise Exception(msg)
         cmd = './AzureAttestSKR -n 123456 -k {0} -c imds -r'.format(KeyEncryptionKeyUrl)
         executor = CommandExecutor(self.logger)
         result = executor.Execute(cmd)
         if result != CommonVariables.process_success:
-            if manage_id:
-                raise Exception ('{0} Encryption managed identity: {1}'.format(msg,manage_id))
-            raise Exception(msg)
+            #SKR is failed trace the logs.
+            self.logger.log('enabling SKR tracing. set SKR_TRACE_ON = 1')
+            os.environ["SKR_TRACE_ON"]="1"
+            result = executor.Execute(cmd)
+            #re-check, to ensure the SKR. 
+            if result != CommonVariables.process_success:
+                raise Exception(msg)
```

Comment:
we want to log the stdout/stderr of the executed cmd. Is msg implicitly capturing that?

Reply:
yes it is captured at stderr:
part of snipped is:
2023/11/29 05:30:57 stdout:
2023/11/29 05:30:57 Tracing is enabled
2023/11/29 05:30:57 Level: Info Tag: AttestatationClientLib ReadAkCertFromTpm:118:Successfully fetched the AK cert from TPM
.....
2023/11/29 05:30:57 stderr:
2023/11/29 05:30:57 Main started
2023/11/29 05:30:57 nonce: 123456
2023/11/29 05:30:57 key_enc_key_url: ...
2023/11/29 05:30:57 akv_credential_source: 0
2023/11/29 05:30:57 op: 3
2023/11/29 05:30:57 Entering Util::ReleaseKey()
2023/11/29 05:30:57 Entering Util::doSKR()
2023/11/29 05:30:57 Entering Util::GetMAAToken()
2023/11/29 05:30:57 Exiting Util::GetMAAToken()